setup.py

```diff
@@ -1,308 +1,105 @@
-            if not static and not windows:
-                self.extensions[0].runtime_library_dirs += [os.path.join(adir, lib_path)]
-
-        # Add the static linking options.
-
-        # Add MSVC specific options.
-        if windows and not msys2:
-            self.extensions[0].extra_link_args.append('/MANIFEST')
-            self.extensions[0].define_macros.append(("MPIR", 1))
-            if not static:
-                self.extensions[0].define_macros.append(("MSC_USE_DLL", None))
-        else:
-            if mpir:
-                if static and mp_found:
-                    self.extensions[0].extra_objects.append(os.path.join(mp_found, lib_path, 'libmpir.a'))
-            else:
-                if static and mp_found:
-                    self.extensions[0].extra_objects.append(os.path.join(mp_found, lib_path, 'libgmp.a'))
-            if static and mpfr_found:
-                self.extensions[0].extra_objects.append(os.path.join(mpfr_found, lib_path, 'libmpfr.a'))
-            if static and mpc_found:
-                self.extensions[0].extra_objects.append(os.path.join(mpc_found, lib_path, 'libmpc.a'))
-
+        self.fast = False
+        self.gcov = False
+        self.vector = False
+        self.mpir = False
+        self.static = False
+        self.gdb = False
 
     def finalize_options(self):
         build_ext.finalize_options(self)
-        gmpy_build_ext.doit(self)
-
-# custom install data in order that data_files
-# get installed together with the .so
-
-class gmpy_install_data(install_data):
-    def finalize_options(self):
-        install_data.finalize_options(self)
-        install = self.distribution.get_command_obj('install')
-        self.install_dir = install.install_purelib
-
-# prepare the extension for building
-
-my_commands = {'clean' : gmpy_clean, 'build_ext' : gmpy_build_ext, 'install_data' : gmpy_install_data}
-
-gmpy2_ext = Extension('gmpy2',
-                      sources=[os.path.join('src', 'gmpy2.c')],
-                      include_dirs=['./src'],
-                      define_macros = defines)
-
-setup(name = "gmpy2",
-      version = "2.1.0a1",
-      author = "Case Van Horsen",
-      author_email = "casevh@gmail.com",
-      license = "LGPL-3.0+",
-      url = "https://github.com/aleaxit/gmpy",
-      description = "gmpy2 interface to GMP/MPIR, MPFR, and MPC for Python 2.6+ and 3.4+",
-      data_files = [('', ['src/gmpy2.pxd']), ('gmpy2', ['src/gmpy2.h'])],
-      classifiers = [
+        if self.fast:
+            _comp_args.append('-DFAST=1')
+        if self.gcov:
+            if ON_WINDOWS:
+                raise ValueError("Cannot enable GCC code coverage on Windows")
+            _comp_args.append('-DGCOV=1')
+            _comp_args.append('-O0')
+            _comp_args.append('--coverage')
+            link_args.append('--coverage')
+            _libs.append('gcov')
+        if self.vector:
+            _comp_args.append('-DVECTOR=1')
+        if self.mpir or ON_WINDOWS:
```

Comment:
This means we can't use `gmp` on windows. You should set `self.mpir = ON_WINDOWS` when you are setting the default values in Line 37 above

Reply:
Forcing gmp on Windows would be another flag which did not exist before. It can be added, but the current behaviour reflects the old setup.py which did not allow gmp on windows either.

Reply:
We can assume `gmp` if not `mpir`. 

> It can be added, but the current behaviour reflects the old setup.py which did not allow gmp on windows either.

This isn't true. Why do you think that it wasn't allowed?

Reply:
The legacy setup.py forced mpir on for windows and MSVC:
```
        # Add MSVC specific options.
        if windows and not msys2:
            self.extensions[0].extra_link_args.append('/MANIFEST')
            self.extensions[0].define_macros.append(("MPIR", 1))
            if not static:
                self.extensions[0].define_macros.append(("MSC_USE_DLL", None))

```

Reply:
Yes, but it didn't force `mpir` on windows and msys2

Reply:
That's fine, will default msys2 to gmp where [that is detected](https://github.com/aleaxit/gmpy/pull/178/files#diff-2eeaed663bd0d25b7e608891384b7298R68). 

The point is it was not possible to force use of gmp on MSVC before, nor is it possible with these changes.

Reply:
Well, with msys2, you can use either gmp or mpir. With MSVC, it's just mpir